model/BookletClassService.php

```diff
@@ -113,11 +113,25 @@ public function getTest(core_kernel_classes_Resource $booklet)
     /**
      * Get the attachment linked to a booklet
      * @param core_kernel_classes_Resource $booklet
-     * @return core_kernel_classes_Resource
+     * @return string
+     * @throws \core_kernel_persistence_Exception
      */
     public function getAttachment(core_kernel_classes_Resource $booklet)
     {
-        return (string) $booklet->getOnePropertyValue($this->getProperty(self::PROPERTY_FILE_CONTENT));
+        /** @var core_kernel_classes_Resource $attachment */
+        $attachment = $booklet->getOnePropertyValue($this->getProperty(self::PROPERTY_FILE_CONTENT));
+
+        if ($attachment instanceof core_kernel_classes_Resource) {
```

Comment:
IMHO it would be slightly more readable wich switch..case

```php
$type = is_object($var) ? get_class($var) : gettype($var);
switch($type) {
 case: core_kernel_classes_Resource::class;
   return $attachment->getUri();
 case: core_kernel_classes_Literal::class;
  ...
 case: 'string';
   ...
 default:
  ...
}
```